### PR TITLE
Various NSURLProtocol/NSURLConnection Fixes

### PR DESCRIPTION
**Pull request by advcsi**
Fix circular reference between NSURLProtocol and NSURLConnection
 Implement missing method in NSURLConnection:
        initWithRequest:delegate:startImmediately:
        start
    File(s) changed:
       Headers/Foundation/NSURLConnection.h
       Source/NSURLConnection.m
       Source/NSURLProtocol.m

### CODE REVIEW

File: Source/NSURLProtocol.m [hunk 1]
```diff
@@ -847,7 +847,6 @@ - (void) startLoading
 	self, @"HEAD",
 	self, @"GET",
 	self, @"POST",
-	self, @"PATCH",
```
Comment (bekkii77) [inline Source/NSURLProtocol.m]: Why did you remove PATCH here?

File: Source/NSURLProtocol.m [hunk 2]
```diff
@@ -1519,8 +1523,8 @@ - (void) _got: (NSStream*)stream
 					  code: 0
 				      userInfo: nil];
 		  [self stopLoading];
-		  [this->client URLProtocol: self
-			   didFailWithError: e];
+		  [this->client URLProtocol: self didFailWithError: e];
+      DESTROY(this->client);
```
Comment (bekkii77) [inline Source/NSURLProtocol.m:1528]: Tabbing?

Comment (advcsi) [reply Source/NSURLProtocol.m:1528]: True - but GNUstep is so messed up between people using spaces and tabs that I'm not going to waste my time on this.

File: Source/NSURLProtocol.m (hunk 1 above)
Comment (advcsi) [reply Source/NSURLProtocol.m]: I noticed this after I pushed.  I am not sure how this happened, as I branched from the testplant branch.  I can manually merge the branch to ensure that this does not get lost.

Comment (bekkii77) [reply Source/NSURLProtocol.m]: I'm nervous that other changes may be lost as well?  Please be careful.

**Review by bekkii77 — APPROVED** [review]
